src/cascadia/WindowsTerminal/IslandWindow.cpp

```diff
@@ -757,7 +759,28 @@ long IslandWindow::_calculateTotalSize(const bool isWidth, const long clientSize
                 if (isCurrentlyDark != _currentSystemThemeIsDark)
                 {
                     _currentSystemThemeIsDark = isCurrentlyDark;
-                    UpdateSettingsRequested.raise();
+                    updateSettings = true;
+                }
+
+                bool isCurrentlyHighContrast = []() {
+                    HIGHCONTRAST hc = { 0 };
+                    hc.cbSize = sizeof(HIGHCONTRAST);
+                    if (SystemParametersInfo(SPI_GETHIGHCONTRAST, sizeof(hc), &hc, 0))
+                    {
+                        return (bool)(hc.dwFlags & HCF_HIGHCONTRASTON);
+                    }
+                    return false;
+                }();
+
+                if (isCurrentlyHighContrast != _currentHighContrastModeState)
+                {
+                    _currentHighContrastModeState = isCurrentlyHighContrast;
+                    updateSettings = true;
+                }
+
+                if (updateSettings)
+                {
+                    UpdateSettingsRequested.raise(_currentHighContrastModeState);

```

Comment:
Why does the `IslandWindow` of all things need to remember whether we were in dark mode or not? Also, why does it result in `ReloadSettings` when none of the settings changed? An alternative idea:
* Add a `bool CascadiaSettings::ReloadingSystemConfig()` function (or similar).
  * It reads the dark-mode-ness and the high-contrast-ness of the OS.
  * It caches the 2 results inside (for instance) `GlobalAppSettings`.
  * If checks if either of 2 have changed and returns `true` if they did, `false` otherwise.
  * The cached values replace any use of `IsSystemInDarkTheme()` inside the settings code (currently in 2 locations).
* Add an alternative to `ReloadSettings()`, e.g. `ReloadingSystemConfigRelatedSettings()` (lol) and call that in `WindowEmperor::_windowRequestUpdateSettings()`.
* In `ReloadingSystemConfigRelatedSettings()` call `_settings.ReloadingSystemConfig()` and check if it returned `true`. If it did: Raise a `SettingsChanged` event, just like we do in `ReloadSettings`.

Reply:
Simplified it even further by getting AccessibilitySettings to work. Woo! 🎉